### ISSUE TITLE
No type error when overloading with int+float

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#410]
The line with the first call to f is accepted by the type checker, even though it should be rejected similar to the second call:

```
from typing import overload

@overload
def f(n: int) -> int:
    return 0

@overload
def f(n: float) -> float:
    return 1.1

f(1) + 'x'   # PROBLEM: No error reported!
f(1.1) + 'x'
```

I believe that this is due to int being duck type compatible with float.

This is related to #39.
